clj/src/slipstream/ui/views/tables.clj

```diff
@@ -980,14 +983,25 @@
        (map (fn[[k v]] [k (-> v :unit_minutes (uc/format-metric-value k))]))
        (into {})))
 
+(defn format-period
+  [frequency start_timestamp]
+  (case (keyword frequency)
+    :daily    (time/format :date start_timestamp)
+    :weekly   (let [dt-plus-6-days (-> start_timestamp time/parse (ctime/plus (ctime/days 6)))
+                    plus-6-days (format/unparse (:date-time format/formatters) dt-plus-6-days)]
+                (str
+                  (time/format :date-short start_timestamp)
+                  " - "
+                  (time/format :date-short plus-6-days)))
+    :monthly  (time/format :month-year start_timestamp)))
+
 (defn- usage-row
-  [{:keys [start_timestamp cloud usage]}]
+  [{:keys [frequency start_timestamp cloud usage]}]
   {:style nil
    :cells [{:type :cell/icon,       :content icons/usage}
-           {:type :cell/date-short, :content start_timestamp}
+           {:type :cell/text,       :content (format-period frequency start_timestamp)}
```

Comment:
Ideally this should be refactored to include e.g. a `:cell/date-period` accepting the `frequency` and the `start_timestamp`.